feat : add wazuh module

MaximeWewer opened the pull request:
# Summary

This PR adds a Wazuh Agent module and enabling full management of Wazuh agent configuration on OPNsense.

# Features

  - Configure Wazuh server connection (address, port, protocol)
  - Manage authentication settings
  - Control log collection and monitoring modules
  - Handle complex OPNsense API structures with custom translation logic

Commit:
feat : add wazuh module

Commit:
feat : add wazuh module

Commit:
chore : update doc

Commit:
chore : update doc

superstes committed:
feat: wazuh-agent - update docs and test-dependencies

superstes committed:
feat: wazuh-agent - update docs and test-dependencies

superstes commented:
Thank you for providing the PR.
I have made some changes to the docs and added the plugin to the test-dependencies.

I'll have to take some time and look at the `modules_utils/main/wazuh_agent.py` as some of the complexity can be handled by the existing abstraction.

Commit:
feat : wazuh refactor

Commit:
feat : wazuh refactor

MaximeWewer commented:
Hi @jiuka and @superstes,

I've cleaned up the Wazuh agent module to properly use the framework's abstraction layers instead of reinventing the wheel.

What I did:
  - Switched from BaseModule to GeneralModule
  - Used FIELDS_TRANSLATE_SPECIAL for nested API fields instead of manual parsing hell
  - Applied the same patterns as webproxy_general.py
  - Auto-generated field lists

The module now plays nicely with the collection's existing patterns instead of being a special case.

MaximeWewer committed:
Merge branch 'latest' into wazuh

Commit:
feat : rework of module based on PR 286

Commit:
feat : rework of module based on PR 286

MaximeWewer commented:
Hi @jiuka and @superstes,

Base on #286 i've rework wazuh module. Thank's for yours PR, now my code is more clean :)

Commit:
feat : update the module to follow best practices like webproxy_general and update descriptions

Commit:
feat : update the module to follow best practices like webproxy_general and update descriptions

jiuka commented on docs/source/modules/wazuh_agent.rst:
```diff
@@ -0,0 +1,179 @@
+Prerequisites
+*************
+
+You need to install the postfix plugin:
```
the Wazuh agent plugin

jiuka commented on docs/source/modules/wazuh_agent.rst:
```diff
@@ -0,0 +1,179 @@
+   :header: "Parameter", "Type", "Required", "Default", "Aliases", "Comment"
+   :widths: 15 10 10 10 10 45
+
+   "enabled","boolean","false","true","","Enable/disable Wazuh Agent"
```
Empty Default or  Aliases field are usually written as `"\-"`

jiuka commented on plugins/modules/wazuh_agent.py:
```diff
@@ -0,0 +1,126 @@
+    module_dependency_error()
+
+# DOCUMENTATION = 'https://opnsense.ansibleguy.net/modules/wazuh_agent.html'
+# EXAMPLES = 'https://opnsense.ansibleguy.net/modules/wazuh_agent.html'
```
The new url for the documentation and examples is https://ansible-opnsense.oxl.app/

jiuka commented on plugins/module_utils/main/wazuh_agent.py:
```diff
@@ -0,0 +1,65 @@
+        'port': {'min': 1, 'max': 65535},
+        'auth_port': {'min': 1, 'max': 65535},
+    }
+    
```
Adding `FIELDS_DIFF_NO_LOG = ['auth_password']`  would probably be advisable to hide the auth_password from the diff.

jiuka commented on tests/1_cleanup.yml:
```diff
@@ -942,6 +942,15 @@
+      register: cl_wazuh
+      failed_when:
+        - cl_wazuh.failed
+        - "'Unsupported' not in cl_wazuh.msg | default('')"
```
Does the 'Unsupported' case apply to the wazuh agent too? I think this is snapshot as they are only supported on zfs.

MaximeWewer replied:
@jiuka Good catch ! 
No, the 'Unsupported' case does not apply to the Wazuh agent module. 
The 'Unsupported' case is specific to modules that depend on ZFS (like snapshots).

jiuka reviewed (commented):
Just a few nitpicks. Otherwise great work :+1:

Commit:
fix : some fixes

Commit:
fix : some fixes

superstes commented:
@MaximeWewer Please sync your fork and update your changes as the collection-namespace was changed!

Commit:
chore: change collection namespace (ansibleguy => oxlorg)

Commit:
chore: change collection namespace (ansibleguy => oxlorg)

Commit:
fix : change collection namespace on missing files

Commit:
fix : copyright

MaximeWewer committed:
Merge branch 'latest' into wazuh

Commit:
fix : copyright

MaximeWewer commented:
Hi @superstes,

I think we are good now :)

MaximeWewer committed:
Merge branch 'latest' into wazuh

superstes commented:
Sorry for the delay - was sick for some time ._.

The implementation looks super - thank you :+1:

superstes committed:
Merge remote-tracking branch 'origin/wazuh' into wazuh

superstes committed:
fix: minor fixes for wazuh_agent module

superstes committed:
fix: lint fixes for wazuh_agent module

superstes commented:
@MaximeWewer FYI: I commented-out the `active_response_fw_alias_ignore` argument inside the tests, as it failed because of a missing alias (I guess)

Have fixed the other issues with execution (some changes from latest were not yet applied)

superstes commented:
~30m
